microscopium/preprocess.py

```diff
@@ -631,8 +663,8 @@ def correct_multiimage_illumination(im_fns, illum, stretch_quantile=0,
     for im in ims_pass2:
         corrected = im / illum
         rescaled = exposure.rescale_intensity(corrected, in_range=corr_range,
-                                              out_range=np.uint8)
-        out = np.round(rescaled).astype(np.uint8)
+                                              out_range=im0.dtype.type)
+        out = np.round(rescaled).astype(im0.dtype)
```

Comment:
This is a very nice improvement. However, I'm concerned about memory blowout — do we really need anything more than 8 bits after the illumination correction?